front/pages/[user]/a/[sId]/execute/index.jsx

```diff
@@ -0,0 +1,403 @@
+import {
+  CheckCircleIcon,
+  ChevronDownIcon,
+  ChevronRightIcon,
+  PlayCircleIcon,
+} from "@heroicons/react/20/solid";
+import { unstable_getServerSession } from "next-auth/next";
+import { useSession } from "next-auth/react";
+import { useState, useEffect, useRef } from "react";
+import TextareaAutosize from "react-textarea-autosize";
+import { SSE } from "sse.js";
+import {
+  ArrayViewer,
+  ObjectViewer,
+  StringViewer,
+} from "../../../../../components/app/blocks/Output";
+import MainTab from "../../../../../components/app/MainTab";
+import AppLayout from "../../../../../components/AppLayout";
+import { ActionButton, Button } from "../../../../../components/Button";
+import { extractConfig } from "../../../../../lib/config";
+import { checkDatasetData } from "../../../../../lib/datasets";
+import { classNames } from "../../../../../lib/utils";
+
+const { URL, GA_TRACKING_ID = null } = process.env;
+
+function preProcessOutput(output) {
```

Comment:
curious to have your opinion on this -- trying to make it nicer to view (if it's a LLM output, then we'll only show the completion. If it's a 1-element array, then we only display that element

Reply:
Also this probably re-runs anytime an output is added so not very optimal

Reply:
Need to dig more to answer will do asap